### DISCUSSION TITLE
Get rid of common "Maven schlagworts"

### DISCUSSION THREAD

**Discussion by cstamas** [support-and-care/maven-support-and-care#17] [Ideas for Funding]
Two common ones are:
* "Maven downloads whole internet". This was successfully addressed by [MPLUGIN-372](https://issues.apache.org/jira/browse/MPLUGIN-372) that before fix was leading to following scenarios: user downloads Maven version X, builds something with it, and then finds out that local repository container tens if not more versions of `maven-core` and it was puzzling to them. What happened is that due this bug, it was not possible to set plugin Maven dependencies to natural "provided" scope, and Maven at runtime did download these, only to skip the downloaded (core provided) artifacts from the plugin classpath, as they were provided by core. We should continue on this path (but mostly all core plugins are today OK).
* "deploying to Central is hard". Another schlagwort common on twitter/reddit/etc. I understand this is not under our (project) control (ASF Maven), but IMHO we CAN provide a lot of help here to end users, and make process no-brainer.

If anyone else aware of any other commonplace, please put it here.

**Comment by hendrikebbers** [comment]
"Maven is slow" against Gradle since Gradle uses Damon processes.

**Comment by rmannibucau** [comment]
Guess this "maven is slow" thing got killed with maven daemon and maven build cache extension.

The only thing where maven can stay theorically slower is the fact it uses phases instead of building a graph at mojo level but community rejected the idea to drop phases for v4 (and v5 is way too far) so guess we'll stay there.

Rest seems comm since in practise maven is not slower than gradle - apache beam did the migration for 8mn on 120mn and lost part of its contributors cause of that.

Not sure about the download the internet, whatever enhancement we do it comes from the design - plugins being isolated for goods.

About central, it is clearly not very hard and will be easier with the new sonatype plugin so have to admit I have a hard time to convert this thread in actions :thinking:

**Reply by hendrikebbers** [reply]
I’m with you that it is not true from a technical point of view. But people still assume it. So education is needed

**Comment by cstamas** [comment]
As @hendrikebbers this is more about proper education.

OTOH, I'd not count as "solved" with new Sonatype plugin, as I have doubts about it (sources are based on old plugin done by me). So same issues stand there, and is not "future proof".

**Reply by cstamas** [reply]
In other words, "publishing to central" is the paramount, the alpha-and-omega of _Maven_. I never considered Maven as CLI only, it is much much more than that (and people keep forgetting that "binary artifact repository" is Maven invention as well).

**Comment by sparsick** [comment]
"Maven configuration is hard to understand / chatty because it uses still XML" is a phrase that I hear a lot in discussion why people dont like using Maven. I recommend using the XML or Maven Support of the IDE.

**Reply by cstamas** [reply]
Maven4 has built-in polyglot capabilities (a la Takari Polyglot https://github.com/takari/polyglot-maven/).
Here is a Maven4 IT that uses POM written in Hocon https://github.com/apache/maven-integration-testing/pull/276/files#diff-8d7362e60d231ad8c5d4b7746873da2855d9cf1fd5aeeca9c143ed942bd94b38

**Reply by hillgert** [reply]
IMHO XML has many advantages due to XML schema is providing help, information, and suggestions. Maybe using more XML Attributes instead of elements could increase some readability and shrinkens the POM file.

**Comment by Bukama** [comment]
My thoughts/ideas:
1) Get rid of "mvn clean install" - but this will almost impossible to change the internet. But still a widespread missthought, same like
2) The (minimal) Java version needed to run is equal to the Java version of your app

I think we can still improve in those things, tbh not really knowing how, except by talking more about it, like we did at the JCON during the last days.

**Comment by ascheman** [comment]
This discussion has been converted to Epic #61 for tracking.

Closing this discussion.
